### ISSUE TITLE
evidence determination for gene-product's involvement with downstream processes

### ISSUE THREAD

**Issue by ukemi** [geneontology/minerva#99]
When we generate complex models, we want to be able to retrieve correct evidence to support inferences or assertions for processes that are far downstream of intitial annotons. This becomes problematic when the evidence for the association between a gene product's function and the evidence that links the two processes together is different. It is compounded when multiple annotons all feed into a complicated downstream pathway. We need to be able to specify which evidence statement goes with which annoton and we would like to be able to put multiple pieces of evidence on some statements (even inferred ones) if there is experimental evidence to support the direct inference between the action of a gene product and a downstream process.

Here are a couple example to look at for next Thursday's meeting:

simple  http://noctua.berkeleybop.org/editor/graph/gomodel:586fc17a00000729
complex  http://noctua.berkeleybop.org/editor/graph/gomodel:5745387b00001750

computational combinatorial evidence used in automatic assertion(ECO:0000246)?

**Comment by balhoff** [comment]
For reference, one of the ideas we had to support this is in this Noctua request: https://github.com/geneontology/noctua/issues/407

Alternatively, curators could simply add redundant direct relations to the model (which would have otherwise been inferred) and add evidence to them. Ideally a reasoner would inform the curator when any inconsistencies are introduced.

**Comment by ukemi** [comment]
I knew there was a related ticket, but I couldn't find it. Thanks Jim.

**Comment by kltm** [comment]
So, post meeting, the idea now is to add an annotation to the evidence individual that is a virtual edge back to the desired GP individual, correct?

**Comment by ukemi** [comment]
Yes. Although I think this is still a compromise in representation, it is the way that we will proceed based on curator input. So the idea is that we will allow an additional annotation to the evidence that will refer to a gene product (could also be a complex) that is used in the model.

**Comment by kltm** [comment]
But a single individual, right? Or do we want allow for the option to have an evidence individual contain a (virtual) edge back to multiple entities to support the evidence.

For @cmungall and @balhoff , instead of virtual edge annotations, which are easy to deal with as they are not "real", would it be conceptually easier to fold this into the evidence refactor or geneontology/noctua#141 and realize them as actual edges?

**Comment by ukemi** [comment]
I think it should go to a single individual. This leaves open the possibility for what we brought up yesterday concerning multiple instances of the same gene-product class being in a model. Although I think this will be very rare, I think when it happens we will need the link to a single individual for the reasons that @balhoff pointed out on the call. That being, that the final annotation is always inferred from an instance representation of a gene product executing some function at the beginning of a chain of relations. If this could all be somehow behind the scenes so curators wouldn't have to think about it unless it becomes necessary, that would be good.

**Comment by vanaukenk** [comment]
Just FYI, I have constructed a subset of my Notch signaling model that, for simplicity, only contains the function annotations for the ligand and receptor. 

It is fully evidenced and still has all of the features we'd like for testing evidence on downstream processes, but being simpler, is easier to dissect and troubleshoot.

http://noctua.berkeleybop.org/editor/graph/gomodel:5941cdbd00000002

The more complete Notch signaling pathway actually may be an interesting use case for potentially having multiple instances of the same gene class in a model since at least three distinct instances of C. elegans Notch activity, localized to: 1) extracellular space, 2) plasma membrane, and 3) nucleus, are involved in the pathway.  In this case, I can imagine having a piece of evidence that supports involvement of all three activities (e.g. mutant phenotype based on a deletion allele) in regulation of a downstream process, as well as evidence that may only support involvement of a specific activity (e.g. mutant phenotype based on a mutation only affecting formation of the transcription factor complex) and I think I'd want to be able to clearly differentiate between the two.

**Comment by cmungall** [comment]
@vanaukenk - thanks for the simpler model

Do we have a list of what we consider to be errors of omission or comission from:
http://noctua.berkeleybop.org/workbench/annpreview/?model_id=gomodel:5941cdbd00000002&barista_token=

**Comment by vanaukenk** [comment]
@cmungall  - No, but I will create one.

**Comment by vanaukenk** [comment]
From 2019-01-31:
Discussed this issue again at hackathon.
Proposal:
For GPAD output, nodes more than 2 steps downstream will get the evidence between the MF and the first BP or the evidence between the MF and the first CC.
Curators can put evidence on as many edges as needed in the actual model.
This needs to be tested with different models that chain BPs and CCs with different relations.
